games-puzzle/kpat/kpat-17.12.1.recipe

```diff
@@ -0,0 +1,145 @@
+COPYRIGHT="2010-2017 KDE Organisation
+			1995 Paul Olav Tvete"
+LICENSE="GNU LGPL v2
+			POT v2"
+REVISION="1"
+SOURCE_URI="https://github.com/KDE/kpat/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="5241e8092bf4820b4ad2a016fed0928ed12815edb050dbb3f5512103aca0f293"
+#PATCHES="kpat-$portVersion.patchset"
+ADDITIONAL_FILES="kpat.rdef.in"
+	
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	kpat$secondaryArchSuffix = $portVersion
+	app:kpat$secondaryArchSuffix
+	lib:libkcardgame$secondaryArchSuffix
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	cmd:dbus_launch
+	lib:libKF5Attica$secondaryArchSuffix
+	lib:libKF5Auth$secondaryArchSuffix
+	lib:libKF5Bookmarks$secondaryArchSuffix
+	lib:libKF5Codecs$secondaryArchSuffix
+	lib:libKF5Completion$secondaryArchSuffix
+	lib:libKF5ConfigCore$secondaryArchSuffix
+	lib:libKF5ConfigWidgets$secondaryArchSuffix
+	lib:libKF5CoreAddons$secondaryArchSuffix
+	lib:libKF5Crash$secondaryArchSuffix
+	lib:libKF5DBusAddons$secondaryArchSuffix
+	lib:libKF5GuiAddons$secondaryArchSuffix
+	lib:libKF5I18n$secondaryArchSuffix
+	lib:libKF5ItemViews$secondaryArchSuffix
+	lib:libKF5JobWidgets$secondaryArchSuffix
+	lib:libKF5KDEGames$secondaryArchSuffix
+	lib:libKF5KIOCore$secondaryArchSuffix
+	lib:libKF5NewStuff$secondaryArchSuffix
+	lib:libKF5Service$secondaryArchSuffix
+	lib:libKF5WidgetsAddons$secondaryArchSuffix
+	lib:libKF5XmlGui$secondaryArchSuffix
+	lib:libQt5Core$secondaryArchSuffix
+	lib:libQt5Gui$secondaryArchSuffix
+	lib:libQt5Qml$secondaryArchSuffix
+	lib:libQt5Quick$secondaryArchSuffix
+	lib:libQt5Widgets$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	extra_cmake_modules >= 5.39
+	haiku${secondaryArchSuffix}_devel
+	devel:libKF5Attica$secondaryArchSuffix
+	devel:libKF5Auth$secondaryArchSuffix
+	devel:libKF5Bookmarks$secondaryArchSuffix
+	devel:libKF5Codecs$secondaryArchSuffix
+	devel:libKF5Completion$secondaryArchSuffix
+	devel:libKF5ConfigCore$secondaryArchSuffix
+	devel:libKF5ConfigWidgets$secondaryArchSuffix
+	devel:libKF5CoreAddons$secondaryArchSuffix
+	devel:libKF5Crash$secondaryArchSuffix
+	devel:libKF5DBusAddons$secondaryArchSuffix
+	devel:libKF5GuiAddons$secondaryArchSuffix
+	devel:libKF5I18n$secondaryArchSuffix
+	devel:libKF5ItemViews$secondaryArchSuffix
+	devel:libKF5JobWidgets$secondaryArchSuffix
+	devel:libKF5KDEGames$secondaryArchSuffix
+	devel:libKF5KIOCore$secondaryArchSuffix
+	devel:libKF5NewStuff$secondaryArchSuffix
+	devel:libKF5Service$secondaryArchSuffix
+	devel:libKF5Solid$secondaryArchSuffix
+	devel:libKF5WidgetsAddons$secondaryArchSuffix
+	devel:libKF5XmlGui$secondaryArchSuffix
+	devel:libQt5Core$secondaryArchSuffix
+	devel:libQt5Gui$secondaryArchSuffix
+	devel:libQt5Qml$secondaryArchSuffix
+	devel:libQt5Quick$secondaryArchSuffix
+	devel:libQt5Widgets$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:cmake
+	cmd:g++$secondaryArchSuffix
+	cmd:gettext$secondaryArchSuffix
+	cmd:make
+	cmd:python2
+	cmd:update_mime_database$secondaryArchSuffix
```

Comment:
This might save you some variables.

From [KDEInstallDirs][0]:

> If `KDE_INSTALL_USE_QT_SYS_PATHS` is set to TRUE before including this module, the default values for some variables are instead queried from Qt5’s qmake (where mentioned in the parentheses above). If not set, **it will default to TRUE if Qt5’s qmake is found** and it’s `QT_INSTALL_PREFIX` is the same as `CMAKE_INSTALL_PREFIX`, otherwise default to FALSE. This variable should NOT be set from within CMakeLists.txt files, instead is intended to be set manually when configuring a project which uses KDEInstallDirs (e.g. by packagers).

```diff
+	cmd:qmake >= 5
```

[0]: https://api.kde.org/ecm/kde-module/KDEInstallDirs.html

Reply:
I'm not sure i understand what you mean by "This might save you some variables."
From KDEInstallDirs:?
for the "cmd:qmake >= 5" i 've never been able to build anything with it when i add >=5 to it, it always says command not found.

Reply:
According to `KDEInstallDirs` documentation (the cmake module that's used by this package). If you have qmake in your `PATH`, it could use qmake to get default values for some variable (namely LIBDIR, PREFIX,...)

Those usually have the package name in there so maybe this suggestion wasn't so good after all. So feel free to leave it out

Reply:
@khallebal, there are around 20 recipes which have `cmd:qmake$secondaryArchSuffix >= 5`, and I have recently built one of them: [CuteMarkEd](https://github.com/haikuports/haikuports/tree/master/app-text/cutemarked)

BTW, if you are running Haiku x86_gcc2 then you really need the `$secondaryArchSuffix`, otherwise you will get a `command not found` error.

Reply:
@fbrosson It's not necessary. As I wrote [earlier][0], qmake is only used to fetch system prefix. This port uses CMake instead

[0]: https://github.com/haikuports/haikuports/pull/2116#discussion_r161420514